File: CPP Unit 06/Static/Q3.cpp
Single File Programming Question
Problem Statement



Spider-Man is testing the strength of his web shooters across New York City. He needs to calculate the strength of multiple web strands using the formula: Strength = Thickness × Length. 



To manage and analyze the web strengths, he stores them in a vector, passes the vector to a function to find the minimum strength, and displays all the strengths along with the minimum value. 



Help him to implement the task using vectors from the Standard Template Library (STL).

Input format :
The first line of input consists of an integer n representing the number of web strands.

The next n pairs of lines consist of a double thickness representing the thickness of the web in millimeters, followed by a double length representing the length of the web in meters.

Output format :
The first line of output prints "Web strengths:".

The subsequent lines print the strength of each web strand with two decimal places, one strength per line.

The last line of output prints "Minimum strength: " followed by the minimum strength value with two decimal places.



Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

1 ≤ n ≤ 100

0.0 < thickness ≤ 100.0

0.0 < length ≤ 1000.0

Strength output is displayed with exactly 2 decimal places

Sample test cases :
Input 1 :
3
2.0
3.0
1.0
5.0
4.0
2.0
Output 1 :
Web strengths:
6.00
5.00
8.00
Minimum strength: 5.00
Input 2 :
2
5.0
2.0
3.0
4.0
Output 2 :
Web strengths:
10.00
12.00
Minimum strength: 10.00




#include <iostream>
#include <vector>
#include <iomanip>
using namespace std;

double calculateStrength(double thickness, double length) {
    return thickness * length;
}

double findMinimum(vector<double> strengths) {
    double minVal = strengths[0];
    for (int i = 1; i < strengths.size(); i++) {
        if (strengths[i] < minVal) {
            minVal = strengths[i];
        }
    }
    return minVal;
}

int main() {
    int n;
    cin >> n;

    vector<double> strengths;

    for (int i = 0; i < n; i++) {
        double thickness, length;
        cin >> thickness >> length;
        double strength = calculateStrength(thickness, length);
        strengths.push_back(strength);
    }

    cout << fixed << setprecision(2);
    cout << "Web strengths:" << endl;

    for (int i = 0; i < strengths.size(); i++) {
        cout << strengths[i] << endl;
    }

    double minStrength = findMinimum(strengths);
    cout << "Minimum strength: " << minStrength << endl;

    return 0;
}

File: Java/Unit 03/Lec 18/Q3.java
Single File Programming Question
Problem Statement



Akash is learning programming and wants to understand the concept of the final keyword. His friend Roshini gave him a task to calculate the area of a rhombus using the final keyword.



Create a base class named Shape with a method calculateArea() that can be overridden by subclasses. Then, create a final class Rhombus that extends Shape and contains private final member variables diagonal1 and diagonal2, representing the lengths of the diagonals. Implement a constructor to initialize these values and override the calculateArea() method to compute the area.



Formula: 

area = (0.5 * diagonal1 * diagonal2).

Input format :
The input consists of two space-separated double values, representing the length of the diagonals.

Output format :
The output prints a double value representing the area of the rhombus, rounded off to two decimal places.



Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

1.0 ≤ diagonals ≤ 170.0

Sample test cases :
Input 1 :
15.3 12.5
Output 1 :
95.63
Input 2 :
1.0 8.6
Output 2 :
4.30
Input 3 :
159.3 170.0
Output 3 :
13540.50
Whitelist
Set 1:
final


import java.util.Scanner;

// Base class
class Shape {

    // Method to calculate area (can be overridden)
    double calculateArea() {
        return 0;
    }
}

// Final class Rhombus
final class Rhombus extends Shape {

    // Final member variables
    private final double diagonal1;
    private final double diagonal2;

    // Constructor to initialize diagonals
    Rhombus(double diagonal1, double diagonal2) {
        this.diagonal1 = diagonal1;
        this.diagonal2 = diagonal2;
    }

    // Override calculateArea method
    @Override
    double calculateArea() {
        return 0.5 * diagonal1 * diagonal2;
    }
}

// Main class
public class Main {

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        double diagonal1 = scanner.nextDouble();
        double diagonal2 = scanner.nextDouble();

        Rhombus rhombus = new Rhombus(diagonal1, diagonal2);

        double area = rhombus.calculateArea();

        System.out.printf("%.2f\n", area);

        scanner.close();
    }
}
